### PR TITLE
 upgrade libraries versions for SonarQube 10.0.0

### PR DESCRIPTION
**Pull request by dedece35**

### CODE REVIEW

**Comment by utarwyn** [comment]
@dedece35 What are the next steps before merging this PR?

**Comment by dedece35** [comment]
@utarwyn ... to have time to check all is ok with dependencies upgrades :(

**Comment by utarwyn** [comment]
JavaScript plugin seems to work fine on my 10.0 SonarQube server with your changes :+1:

**Comment by dedece35** [comment]
Check Java / python / php plugins :
- ✅ local unit tests
- ✅ **SonarQube 10.0.0** (JDK 17) + build JARs with **JDK 17** + test-projects **JDK17**
- ✅ ??? **SonarQube 10.0.0** + build JARs with **JDK 11** + test-projects **JDK11**
- ✅ **SonarQube 9.9.1** (JDK 17) + build JARs with **JDK 17** + test-projects **JDK17**
- ✅ ???  **SonarQube 9.9.1** (JDK 17) + build JARs with **JDK 11** + test-projects **JDK11**
- **SonarQube 9.4** (JDK 17) + build JARs with **JDK 17** + test-projects **JDK17**
- **SonarQube 9.4** (JDK 17) + build JARs with **JDK 11** + test-projects **JDK11**

**Comment by dedece35** [comment]
**Conclusion of tests**
for me, my modifications are ok only for Sonarqube >= 9.9 and not versions under.
Thus, if we want to keep modifications (upgrade of libraries), we should release a new major version.
Otherwise, I can investigate more to detect which library is the root cause, but we won't upgrade this library in the future until we accept this incompatibility upgrade.
What do you thin about it, @glalloue, @utarwyn, @jycr, @MP-Aubay, @jhertout, @olegoaer, @mdubois81 ?

**Comment by utarwyn** [comment]
According to the SonarQube website, the latest LTS available is 9.9, and they strongly encourage users to upgrade to the latest available version, so we might decide to set this version as the minimum. **BUT** as a majority of our users are companies, it makes sense not to rush updates if maintenance on our side isn't too complex.

I built the plugins based on your branch and put them in my SonarQube 9.8 server (without Docker — sorry I don't have SonarQube 9.4 anymore), and **the plugins load and work fine** ✅.

image

➡️ @dedece35 Maybe an error with your Docker installation ? It seems to be a file permission issue.

Just for information, an LTS is generally supported for 18 months. See more here: https://endoflife.date/sonar

**Comment by utarwyn** [comment]
@dedece35 Java plugin works well on my SonarQube 9.8 and rules are displayed correctly when I analyze Java test project

image

**Comment by dedece35** [comment]
@utarwyn thank you for tests ! I found the problem.
I use Rancher Desktop to manage docker locally ...
I've just tested with Docker Desktop and all is OK with 9.4 et 9.8 and still not with Rancher Desktop :( 
Thanks again !

File: pom.xml [hunk 1]
```diff
@@ -50,21 +50,23 @@
...
+
+        <sonar-analyzer-commons.version>2.5.0.1358</sonar-analyzer-commons.version>
 
-        <sonar-plugin-api-impl.version>9.7.1.62043</sonar-plugin-api-impl.version>
-        <sonar-analyzer-commons.version>2.1.0.1111</sonar-analyzer-commons.version>
         <sonar-packaging.version>1.21.0.505</sonar-packaging.version>
         <sonar.skipDependenciesPackaging>true</sonar.skipDependenciesPackaging>
 
+
+        <jacoco.version>0.8.10</jacoco.version>
```
Comment (utarwyn) [inline pom.xml:69]: I have the impression that this variable is not used?
Further down, the jacoco plugin always uses a hard-coded version.

Comment (dedece35) [reply pom.xml:69]: you're rigth ! I correct it.

**Review by utarwyn — APPROVED** [review]

**Pull request merged**